### DISCUSSION TITLE
rtw_tx_pwr_idx_override vs iwconfig txpower

### DISCUSSION THREAD

**Discussion by XBeg9** [svpcom/wfb-ng#273] [Questions] [answered]
Hi, first of all thanks for the amazing project. Could you please help me understand, while `rtw_tx_pwr_idx_override` states in the docs it's going to set power index and force it, do I still need to set `iwconfig txpower` ?
![photo_2023-01-19 09 51 28](https://user-images.githubusercontent.com/872907/213385049-e3ac8333-a3e8-41cd-ba4f-8e7fc2c29667.jpeg)

Thanks

**Comment by svpcom** [comment, marked as answer]
No, you don't need to use `iwconfig txpower` for 8812au cards. `rtw_tx_pwr_idx_override` module parameter completely overrides card tx power management

**Reply by XBeg9** [reply]
Thanks!

**Reply by gghyoo** [reply]
@svpcom  But if I want to change the tx power at run time , can I use the `iwconfig txpower`, or something other commands.

**Reply by svpcom** [reply]
No, need to use rtw_tx_pwr_idx_override module parameter

**Reply by gghyoo** [reply]
I mean, already used insmod with rtw_tx_pwr_idx_override module parameter, but after that, I need to change the tx power depends on other situations.

**Reply by svpcom** [reply]
Try `echo XX > /sys/module/88XXau_wfb/parameters/rtw_tx_pwr_idx_override` during runtime, but I'm not sure that it will change tx power. Need to do some tests (for example measure consumed current)

**Reply by tpwrules** [reply]
I checked with a spectrum analyzer and this does not change power during runtime. You have to rmmod and then insmod to change the power.

**Reply by svpcom** [reply]
But if you need runtime power management then you can try to replace `registrypriv.RegTxPowerIndexOverride` with `(rtw_tx_pwr_idx_override > MAX_POWER_INDEX ? MAX_POWER_INDEX : rtw_tx_pwr_idx_override)` in `hal/rtl8812a/rtl8812a_phycfg.c`

**Reply by tpwrules** [reply]
I was going to try but wasn't sure if that was enough, I needed to look at the code more closely. The power control in the driver is complicated it seems.
